src/library_syscall.js

```diff
@@ -792,7 +792,11 @@ var SyscallsLibrary = {
     copyString('nodename', 'emscripten');
     copyString('release', '1.0');
     copyString('version', '#1');
-    copyString('machine', 'x86-JS');
+#if MEMORY64 == 1
```

Comment:
I think you can just do `#if MEMORY64`

Reply:
That would also make it set if `MEMORY64 == 2`, is that what we want?  My understanding is that mode 2 was still effectively running on wasm32.

Reply:
Oh I forgot about that mode.   In that case you are right.